docs/advanced/atomic-swap.md

```diff
@@ -0,0 +1,73 @@
+# Atomic Swaps
+
+## Overview
+
+A cross-chain swap is a trade between two users of different cryptocurrencies. For example, one party may send Decred to a second party's Decred address, while the second party would send Bitcoin to the first party's Bitcoin address. However, as the blockchains are unrelated and transactions can not be reversed, this provides no protection against one of the parties never honoring their end of the trade. One common solution to this problem is to introduce a mutually-trusted third party for escrow. An atomic cross-chain swap solves this problem without the need for a third party.
+
+Atomic swaps involve each party paying into a contract transaction, one contract for each blockchain. The contracts contain an output that is spendable by either party, but the rules required for redemption are different for each party involved.
+
+One party (called counterparty 1 or the initiator) generates a secret and pays the intended trade amount into a contract transaction. The contract output can be redeemed by the second party (called counterparty 2 or the participant) as long as the secret is known. If a period of time (typically 48 hours) expires after the contract transaction has been mined but has not been redeemed by the participant, the contract output can be refunded back to the initiator's wallet.
+
+
+## Prerequisites
+
+In order to perform an on-chain atomic swap between 2 cryptocurrencies, there are several prerequisites. Both chains must support:
+
+ * branched transaction scripts
+ * the same hash algorithm in both chains’ transaction scripts
+ * signature checks in transaction scripts
+ * CheckLockTimeVerify or CheckSequenceVerify (“CLTV” and “CSV” for short) in transaction scripts
+
+The hash function in the scripting language is different from the function used for mining. For example, Decred supports blake256, sha256, ripemd160, and sha1.
+The CLTV and the CSV is for performing a timelock.
+
+
+## Decred-compatible cross-chain atomic swapping
```

Comment:
This list is going to have to stay in sync with decred/atomicswap.

perhaps drop the list and point the user to https://github.com/decred/atomicswap/ for the current list?

Reply:
So I should leave out all that paragraph and just redirect to https://github.com/decred/atomicswap/ ? Because I've just added the list, but I can delete it.

Reply:
Ok, now the compatible currencies section only redirects to Github.